### ISSUE TITLE
ohh javascript how dare you !!

### ISSUE THREAD

**Issue by panacist** [TeamNewPipe/website#176]
i think you should guys redesign the website  using a different mindset !
mainly "anything that you can do without javascript, just dooo it !" and for reasons :
first it is simpler and page will load much faster (it takes me years to fully load the front page using tor browser) 
second give me one reason why i don't have the right to see pictures on your website without using javascript ?
third and not last (javascript issues never end) your website isn't usable from my terminal or mobile phone where javascript is disabled i can't see the menu nor i can see the features of your app or how it looks !

tl;dr : javascript sucks, you can do and should do better guys! and if you are using a static page generator find a better template that uses javascript less!

**Comment by B0pol** [comment]
Well do a PR and we'll eventually merge it.

**Comment by panacist** [comment]
@B0pol i read it we'll haha
okay then i will work on it on my free time hope you will like it!

**Comment by B0pol** [comment]
@panacist don't change the whole website though.

Having a working menu without JavaScript could be a first step, and it will greatly improve the user experience.

**Comment by opusforlife2** [comment]
Make sure you regularly post screenshots of what it looks like with your changes. B)

**Comment by panacist** [comment]
@B0pol i feel like changing everything but i will need to sketch a desgn first so yup
@opusforlife2 sure thing !

**Comment by opusforlife2** [comment]
> i feel like changing everything

Slow down, there, buster. Your changes might be rejected, keep in mind, so it's better to do it a little at a time.

**Comment by TheAssassin** [comment]
> mainly "anything that you can do without javascript, just dooo it !" and for reasons :
> first it is simpler and page will load much faster (it takes me years to fully load the front page using tor browser)

"Loading faster" is not based on any facts. JavaScript based features don't really delay anything. The only minor issue might be that the JQuery script is loaded in the `<header>`, but that'll slow down your download as well even if you have JS disabled. Please create a separate issue about this.

> second give me one reason why i don't have the right to see pictures on your website without using javascript ?

You see *every* regular image, nothing is loaded with JavaScript there. As you mention Tor Browser, I assume you have the most strict setting activated. This setting deactivates HTML5 video playback, and that is likely why you don't see the video be played. The player itself does not depend on JS at all, it's in fact your browser settings that prevent you from seeing this. This is a completely different topic.
Feel free to create an issue about introducing a static image fallback for the video.

> third and not last (javascript issues never end) your website isn't usable from my terminal or mobile phone where javascript is disabled i can't see the menu nor i can see the features of your app or how it looks !

Terminal browsers are the kind of client nobody writes websites for specifically anyway. Compatibility to those will never be a topic. If you don't like to browse the web with a regular browser, maybe don't use the web at all. It's like demanding a radio station to set up some huge speakers so you can follow their program without having to have a receiver yourself.

Disabling JavaScript entirely is not expected any more, and it hasn't a topic so far. We'd probably consider merging a `<noscript>` fallback for that menu. It's, as far as I can tell, the only thing that won't work other than the comments system in the blog. The features stuff is less critical, but if you insist, a fallback might be appropriate there as well.

Please reword your original issue and the issue title if this is what you want to reach. Right now it just looks like one of these "written in rage mode" issues, and won't be taken serious at all.

> i think you should guys redesign the website using a different mindset !

You should probably begin changing your own mindset. Reading and understanding your issue was quite strenuous due to the lack of any serious formatting, proper English grammar/spelling/case sensitivity and the very demanding language. Consider writing better issues which focus on the important aspects, and your suggestions will likely be taken more serious.

**Comment by panacist** [comment]
@TheAssassin sorry i thought i was polite i didn't mean to demand anything from you guys really sorry and if you believe the current website loads fine without javascript then excuse me because it doesn't load properly for me.
to just go through your arguments one by one :
well actually disabling js theoritically have to make the page loads faster. less code less delays this is something i think  should be true no matter what age we are living in.
javascript is something i don't like personally due to various reasons but this is not a topic to debate in comments and i think you are right a noscript fallback should be fine.
and regarding changin my mindset again i didn't mean to be the rude guy in the prison yard i was just expressing my opinions openly and wasn't intentionally trying to hurt anyone's feelings.
grammar/case sensitivity/spelling those are irrelivant but the content should be?
and sorry for the last time Assassin !!
@opusforlife2 sounds like assassin doesn't want me to change anything, so everything will be rejected haha.

**Comment by opusforlife2** [comment]
> grammar/case sensitivity/spelling those are irrelivant but the content should be?

How do we consume content? That's right, through language.

Everyone reads a sentence differently based on their own personality, current frame of mind, and so many other factors. If you write in a neutral way that is clear to everyone, then there is a far lower chance of someone misinterpreting you or taking offence.

**Comment by panacist** [comment]
ahh and the terminal thing : look assassin i just hate web2.0 i wish all websites go back to the first days of the web no pop up ads, no websites with app size front ends , no ajax bullshit...etc
when designing or building any technology that you want to be successful i guess thinking for the minorities and having support for them will be a turning point !
okay terminals are bad but how about blind people who are using screen readers?
should you drop support from those as well? see we are getting into more ethical issues here.
the web as i see it should be a few pages with simple design.. simplicity is a key feature in every complex system

**Comment by opusforlife2** [comment]
> sounds like assassin doesn't want me to change anything, so everything will be rejected haha.

If your changes are based on sound logic and can be verified as objectively better, then no one has a reason to reject them.

**Comment by panacist** [comment]
@opusforlife2 i do agree that language is important but if i keep worrying about grammar and spelling then i will never ever convey the message (content) which is the important part of it all.
i just hope i wasn't misunderstood because of my opposition to javascript. 
> If your changes are based on sound logic and can be verified as objectively better, then no one has a reason to reject them.

i know i was just kidding of course

**Comment by opusforlife2** [comment]
> i do agree that language is important but if i keep worrying about grammar and spelling then i will never ever convey the message (content) which is the important part of it all.

That's why the concept of revising drafts and editing exists. 🤭
